plugins/kinetic-dp/fu-kinetic-dp-connection.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2016 Mario Limonciello <mario.limonciello@dell.com>
+ * Copyright (C) 2017 Peichen Huang <peichenhuang@tw.synaptics.com>
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2021 Jeffrey Lin <jlin@kinet-ic.com>
+ * Copyright (C) 2022 Hai Su <hsu@kinet-ic.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <gio/gio.h>
+
+#define FU_TYPE_KINETIC_DP_CONNECTION (fu_kinetic_dp_connection_get_type())
+G_DECLARE_FINAL_TYPE(FuKineticDpConnection,
+		     fu_kinetic_dp_connection,
+		     FU,
+		     KINETIC_DP_CONNECTION,
+		     GObject)
+
+FuKineticDpConnection *
+fu_kinetic_dp_connection_new(gint fd);
+
+gboolean
+fu_kinetic_dp_connection_read(FuKineticDpConnection *self,
+			      guint32 offset,
+			      guint8 *buf,
+			      gssize length,
+			      GError **error);
+gboolean
+fu_kinetic_dp_connection_write(FuKineticDpConnection *self,
+			       guint32 offset,
+			       const guint8 *buf,
+			       gssize length,
+			       GError **error);
+
+/* native DPCD fields defined in DP spec */
+#define DPCD_ADDR_IEEE_OUI	    0x00300
+#define DPCD_SIZE_IEEE_OUI	    3
+#define DPCD_ADDR_BRANCH_DEV_ID_STR 0x00503
+#define DPCD_SIZE_BRANCH_DEV_ID_STR 6
+#define DPCD_ADDR_BRANCH_FW_SUB	    0x00508
+#define DPCD_SIZE_BRANCH_FW_SUB	    1
+#define DPCD_ADDR_BRANCH_HW_REV	    0x00509
+#define DPCD_SIZE_BRANCH_HW_REV	    1
+#define DPCD_ADDR_BRANCH_FW_MAJ_REV 0x0050A
+#define DPCD_SIZE_BRANCH_FW_MAJ_REV 1
+#define DPCD_ADDR_BRANCH_FW_MIN_REV 0x0050B
+#define DPCD_SIZE_BRANCH_FW_MIN_REV 1
```

Comment:
I feel like these probably should be libfwupdplugin defines somewhere instead

Reply:
Yes and no. we could certainly put something like

xxx

...somewhere -- although I'm not entirely sure. The Kinetic hardware is unusual and seems to steal 0x00508 from a non-standard location -- notice `0x00503 + 6 != 0x00508` :)